src/globals.c

```diff
@@ -543,3 +545,10 @@ void free_globals(void)
 		bundle_to_add = NULL;
 	}
 }
+
+void save_cmd(int argc, char **argv)
+{
+	for (int i = 0; i < argc; i++) {
+		string_or_die(&swupd_cmd, "%s %s", swupd_cmd, argv[i]);
```

Comment:
I see what you mean by leaking the previous swupd_cmds and I'll fix that - but I'm not seeing it only save the last arg. When I test this it saves the whole command the way I expect (with an extra space in front from the first iteration where swupd_cmd is empty).

When I print out suwpd_cmd after saving it I see this:

```
  /path/to/lt-swupd update -S /path/to/state -p /path/to/swupd-client/test/functional/update/re-update-required/target-dir -u file:///path/to/web-dir -C /path/to/Swupd_Root.pem -I
```